common/faceting.js

```diff
@@ -452,12 +437,39 @@
                         }
                     }
 
+                    /**
+                     * Given an object with `date` and `time`, it will turn it into timestamp reprsentation of datetime.
+                     * @param  {object} obj The datetime object with `date` and `time` attributes
+                     * @return {string} timestamp in submission format
+                     * NOTE might return `null`
+                     */
+                    function dateTimeToTimestamp(obj) {
+                        if (!obj) return null;
+                        var ts = obj.date + obj.time;
+                        return moment(ts, dataFormats.date + dataFormats.time24).format(dataFormats.datetime.submission);
+                    }
+
+                    /**
+                     * Given a string representing timestamp will turn it into an object with `date` and `time` attributes
+                     * @param  {string} ts timestamp
+                     * @return {object} an object with `date` and `time` attributes
+                     * NOTE might return `null`
+                     */
+                    function timestampToDateTime(ts) {
+                        if (!ts) return null;
```

Comment:
Thinking about this and what I said to you in slack earlier, should the `null` case return the empty object?
```
obj = {
  date: null,
  time: null
}
```